### ISSUE TITLE
ansible 2.9.12 cron tasks clears the rest of the file

### ISSUE THREAD

**Issue by gryphius** [ansible/ansible#71460]
##### SUMMARY
executing a `cron` tasks with ansible 2.9.12 clears all other jobs/environment variables present in the specified `cron_file`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
cron

##### ANSIBLE VERSION

```
ansible 2.9.12
  config file = None
  configured module search path = ['/Users/schacher/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/schacher/venv/ansible/lib/python3.8/site-packages/ansible
  executable location = /Users/schacher/venv/ansible/bin/ansible
  python version = 3.8.5 (default, Jul 21 2020, 10:48:26) [Clang 11.0.3 (clang-1103.0.32.62)]
```

##### CONFIGURATION

```paste below
(no output)
```

##### OS / ENVIRONMENT

 * Mac OSX 10.15.6
 * Python 3.8.5

##### STEPS TO REPRODUCE

 * write a simple playbook with two cron tasks that write to the same file. Example below
 * execute the playbook
 * with ansible 2.9.12 only the second job is present at the end of playbook execution. downgrading to 2.9.11 and executing the playbook again fixes the problem, both jobs are visible

```yaml
- hosts: all
  connection: local
  tasks:
    - name: job 1
      cron:
        name: "job 1"
        cron_file: demo
        job: "echo '1'"
        user: "demo"

    - name: job 2
      cron:
        name: "job 2"
        cron_file: demo
        job: "echo '2'"
        user: "demo"
```

##### EXPECTED RESULTS
(starting with no  existing file `/etc/cron.d/demo`. This is performed with ansible 2.9.11 which doesn't show this bug)
```
TASK [job 1] ******************************************************************************************************************************************
--- before: /etc/cron.d/demo
+++ after: /etc/cron.d/demo
@@ -0,0 +1,2 @@
+#Ansible: job 1
+* * * * * demo echo '1'

changed: [localhost]

TASK [job 2] ******************************************************************************************************************************************
--- before: /etc/cron.d/demo
+++ after: /etc/cron.d/demo
@@ -1,2 +1,4 @@
 #Ansible: job 1
 * * * * * demo echo '1'
+#Ansible: job 2
+* * * * * demo echo '2'

changed: [localhost]

PLAY RECAP ********************************************************************************************************************************************
localhost                  : ok=3    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```

```
cat /etc/cron.d/demo
#Ansible: job 1
* * * * * demo echo '1'
#Ansible: job 2
* * * * * demo echo '2'
```

##### ACTUAL RESULTS
(starting with no  existing file `/etc/cron.d/demo`)

job 2 removes job1, at the end, only job2 is visible in the file.

```
TASK [job 1] ******************************************************************************************************************************************
--- before: /etc/cron.d/demo
+++ after: /etc/cron.d/demo
@@ -0,0 +1,2 @@
+#Ansible: job 1
+* * * * * demo echo '1'

changed: [localhost]

TASK [job 2] ******************************************************************************************************************************************
--- before: /etc/cron.d/demo
+++ after: /etc/cron.d/demo
@@ -1,2 +1,2 @@
-#Ansible: job 1
-* * * * * demo echo '1'
+#Ansible: job 2
+* * * * * demo echo '2'

changed: [localhost]

PLAY RECAP ********************************************************************************************************************************************
localhost                  : ok=3    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

```
cat /etc/cron.d/demo
#Ansible: job 2
* * * * * demo echo '2'
```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/cron.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cron.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mkrizek** [comment]
Duplicate of https://github.com/ansible/ansible/issues/71368
